api/post.go

```diff
@@ -680,10 +687,20 @@ func sendNotifications(c *Context, post *model.Post, team *model.Team, channel *
 								msg.DeviceId = strings.TrimPrefix(session.DeviceId, model.PUSH_NOTIFY_ANDROID+":")
 							}
 
-							if channel.Type == model.CHANNEL_DIRECT {
-								msg.Message = senderName + userLocale("api.post.send_notifications_and_forget.push_message")
+							if *utils.Cfg.EmailSettings.PushNotificationContents == model.FULL_NOTIFICATION && model.IsValidHttpsUrl(*utils.Cfg.EmailSettings.PushNotificationServer) {
```

Comment:
We should allow http url too.  For example we support http://push-test.mattermost.com for testing and as a free service for the community.